### PR TITLE
Fix for when to add `OpenQASM3ParseFailure` to diagnostic

### PR DESCRIPTION
**Pull request by SooluThomas**
## Summary

Without the change made in this PR, a list of diagnostics with all kinds of `diagLevel`s are added to category `OpenQASM3ParseFailure`. This PR changes this behavior to just having `Error` and `Fatal` severity `diagLevel` in the  `OpenQASM3ParseFailure` category.

### CODE REVIEW

File: lib/Frontend/OpenQASM3/OpenQASM3Frontend.cpp [hunk 1]
```diff
@@ -202,7 +202,7 @@ llvm::Error qssc::frontend::openqasm3::parse(
         llvm::errs() << level << " while parsing OpenQASM 3 input\n"
                      << errMsg.str();
 
-        if (diagnosticCallback_) {
+        if (diagnosticCallback_ and diagLevel == qssc::Severity::Error) {
```
Comment (SooluThomas) [inline lib/Frontend/OpenQASM3/OpenQASM3Frontend.cpp:205]: Should we add other `diagLevel`s to any other kind of category

**Review by taalexander — DISMISSED** [review]
Can you please add a test that warnings are not reported as an exception but errors are?

File: lib/Frontend/OpenQASM3/OpenQASM3Frontend.cpp (hunk 1 above)
Comment (SooluThomas) [reply lib/Frontend/OpenQASM3/OpenQASM3Frontend.cpp]: Added ICE severity to diagnostics in https://github.com/openqasm/qe-compiler/pull/296/commits/9c6d6a5e2b17281426c27fdb4df63b791cfc727d

**Comment by SooluThomas** [comment]
yes, will do after merge 🎉

**Pull request merged**